src/utils/parse-vitest-fn-call.ts

```diff
@@ -418,6 +418,17 @@ export const resolveScope = (
         if (key?.name === identifier) return 'testContext'
       }
 
+      /** if detect test function is created with `.extend()` */
+      if (
+        def.node.type === AST_NODE_TYPES.VariableDeclarator &&
+        def.node.id.type === AST_NODE_TYPES.Identifier &&
+        TestCaseName.hasOwnProperty(def.node.id.name) &&
+        def.node.init?.type === AST_NODE_TYPES.CallExpression &&
+        def.node.init.callee.type === AST_NODE_TYPES.MemberExpression &&
+        isIdentifier(def.node.init.callee.property, 'extend')
```

Comment:
Ideally instead of relying on variable name and the fact that `.extend()` is called on it we should check that the variable is actually imported from `vitest`. I couldn't figure it out quickly, so this seems to be a good enough solution.